topic-configuration/README.adoc

```diff
@@ -0,0 +1,689 @@
+|===
+h|Editable
+|Yes
+
+h|Type
+|List
+
+h|Default value
+|`Delete`
+
+h|Supported values
+|`Delete`, `Compact`, `Compact, Delete`
+
+h|Kafka property name
+|`cleanup.policy`
+|===
+--
+
+Delete retention time::
++
+--
+Delete retention time is the amount of time that deletion tombstone markers are retained if the topic is compacted.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`86400000 ms` (1 day)
+
+h|Kafka property name
+|`delete.retention.ms`
+|===
+--
+
+Minimum cleanable dirty ratio::
++
+--
+Minimum cleanable dirty ratio is the ratio of entries in the log that can be compacted versus entries that cannot be compacted. When this ratio is reached, the eligible messages in the log are compacted. By default, the ratio is `0.5` or 50%, meaning that messages are compacted after at least half of the log messages are eligible. This property applies only when the topic cleanup policy is set to `Compact` or `Compact, Delete`.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Double
+
+h|Default value
+|`0.5` (50%)
+
+h|Kafka property name
+|`min.cleanable.dirty.ratio`
+|===
+--
+
+Minimum compaction lag time::
++
+--
+Minimum compaction lag time is the minimum time a message remains uncompacted in a log. This property applies only when the topic cleanup policy is set to `Compact` or `Compact, Delete`.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`0 seconds`
+
+h|Kafka property name
+|`min.compaction.lag.ms`
+|===
+--
+
+=== Replication
+
+These properties control the behavior of your replicas. Each of these properties impacts every replica created in the topic.
+
+Unclean leader election::
++
+--
+Unclean leader election allows a follower replica that is not in sync with the partition leader to become the leader of the partition. This property provides a way to retain at least partial data if partition leaders are jeopardized. However, this property can lead to data loss, so it is disabled by default.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Boolean
+
+h|Default value
+|`Disabled`
+
+h|Kafka property name
+|`unclean.leader.election.enable`
+|===
+--
+
+=== Cleanup
+
+These properties control the cleanup processing of the log.
+
+Log segment size::
++
+--
+Log segment size is the size of the log segment files. Log processing actions, such as deletion and compaction, operate on log segments, so a larger setting results in fewer files but less frequent log processing.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`1 GB`
+
+h|Kafka property name
+|`segment.bytes`
+|===
+--
+
+Segment time::
++
+--
+Segment time is the amount of time after which the current log segment is rolled even if the segment file is not full. This property ensures that the topic cleanup policy can delete or compact old messages as needed.
```

Comment:
Current log segment will be the _active_log segment

Reply:
I don't quite get this sentence: _This property ensures that the topic cleanup policy can delete or compact old messages as needed._ It means that the active segment is rolled every 7 days. Messages older than that are deleted or compacted according to policy.

Reply:
Yes, the segment size doesn't determine the timing of cleanup policy. It was the _ensures_ I wasn't sure about. I think the point being made here is that by rolling the active log we _can_ apply policy on older log segments.

Reply:
Ack, I see. @Simon, what do you think for making this clear/more accurate? ^^

Reply:
Compaction and deletion are not done on the active log segment AFAIU. This setting says that "even though the `retention.bytes` (for example) hasn't been reached, roll this segment (create a new active segment) so that it can be compacted or deleted". I don't have a problem with the current text but you could change it to `Segment time is the amount of time after which the current log segment is rolled even if the segment file is not full. This enables the segment to be deleted or messages compacted as needed".`

Reply:
@sjwoodman, ah, I see. How about this then. That work?

> Segment time is the amount of time after which the current log segment is rolled even if the segment file is not full. This property enables the segment to be deleted or compacted as needed, even if the log retention limits have not yet been reached.

Reply:
LGTM